sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,158 @@
+    new_reference_pixel = affine_params.scale * affine_params.rotation_matrix @ old_reference_pixel + affine_params.translation
+    reference_coord = reference_map.wcs.pixel_to_world(new_reference_pixel[0],new_reference_pixel[1])
+    Txshift = reference_coord.Tx - target_map.reference_coordinate.Tx
+    Tyshift = reference_coord.Ty - target_map.reference_coordinate.Ty
+
+    # Create a new map with the updated metadata
+    fixed_map = target_map.shift_reference_coord(Txshift, Tyshift)
+    fixed_map.meta["PC1_1"] = new_pc_matrix[0, 0]
+    fixed_map.meta["PC1_2"] = new_pc_matrix[0, 1]
+    fixed_map.meta["PC2_1"] = new_pc_matrix[1, 0]
+    fixed_map.meta["PC2_2"] = new_pc_matrix[1, 1]
+    fixed_map.meta['cdelt1'] = (target_map.scale[0] / affine_params.scale[0]).value
+    fixed_map.meta['cdelt2'] = (target_map.scale[1] / affine_params.scale[1]).value
+    return fixed_map
+
+
+def _warn_user_of_separation(reference_map, target_map):
+    """
+    Issues a warning if the separation between the ``reference_map`` and
+    ``target_map`` is large.
+
+    Parameters
+    ----------
+    reference_map : `sunpy.map.Map`
+        The reference map to which the target map is to be coaligned.
+    target_map : `sunpy.map.Map`
+        The target map to be coaligned to the reference map.
+    """
+    # Maximum angular separation allowed between the reference and target maps
+    tolerable_angular_separation = 1*u.deg
+    ref_coord = SkyCoord(reference_map.observer_coordinate)
+    target_coord = SkyCoord(target_map.observer_coordinate)
+    if astropy.__version__ >= "6.1.0":
+        angular_separation = ref_coord.separation(target_coord, origin_mismatch="ignore")
+    else:
+        angular_separation = ref_coord.separation(target_coord)
+    if angular_separation > tolerable_angular_separation:
+        warnings.warn(
+            "The angular separation between the reference and target maps is large. "
+            "This could cause errors when calculating shift between two "
+            "images. Please make sure the maps are close in space.",
+            SunpyUserWarning,
+            stacklevel=3,
+        )
+    # Calculate time difference and convert to separation angle
+    ref_time = reference_map.date
+    target_time = target_map.date
+    time_diff = abs(ref_time - target_time)
+    time_separation_angle = differential_rotation(time_diff.to(u.day), reference_map.center.Tx, model='howard')
+    if time_separation_angle > tolerable_angular_separation:
+        warnings.warn(
+            "The time difference between the reference and target maps in time is large. "
+            "This could cause errors when calculating shift between two "
+            "images. Please make sure the maps are close in time.",
+            SunpyUserWarning,
+            stacklevel=3,
+        )
+
+
+def coalign(reference_map, target_map, method='match_template'):
```

Comment:
It seems like a bad idea to not have any provision for passing potential keyword parameters to the coalignment methods